drivers/sensor/vl53l0x/vl53l0x.c

```diff
@@ -117,7 +119,7 @@ static int vl53l0x_setup_single_shot(const struct device *dev)
 
 	ret = VL53L0X_SetLimitCheckValue(&drv_data->vl53l0x,
 					 VL53L0X_CHECKENABLE_SIGNAL_RATE_FINAL_RANGE,
-					 VL53L0X_SETUP_SIGNAL_LIMIT);
+					 config->signal_limit_ratio * 0.001 * 65536);
```

Comment:
Hi @wuguanghai45 please don't use floating point in the driver. If needed use integer division instead.